### PR TITLE
drop as_ prefix from as_X_client from storage crates

### PR DESCRIPTION
**Pull request by bmc-msft**
Per discussion in #676, this renames the various as_X_client to X_client

### CODE REVIEW

**Review by rylev — APPROVED** [review]
This is definitely better so we should merge it. We might need to rethink the client hierarchy within the storage crates though...

File: sdk/data_tables/src/clients/entity_client.rs [hunk 1]
```diff
@@ -137,17 +137,17 @@ mod integration_tests {
     }
 
     fn get_emulator_client() -> Arc<TableServiceClient> {
-        let storage_account = StorageAccountClient::new_emulator_default().as_storage_client();
+        let storage_account = StorageAccountClient::new_emulator_default().storage_client();
         storage_account
-            .as_table_service_client()
+            .table_service_client()
```
[the remark this answers is not in the document]
Comment (bmc-msft) [reply sdk/data_tables/src/clients/entity_client.rs:142]: One of the things that could stand clearing up is this issue.  I understand the underlying intent in this implementation, but it's confusing.  Currently, with a StorageAccount.table_service_client gets you a TableServiceClient.

TableServiceClient lets you:
1. list tables
2. get a TableClient for a specified table.

TableClient lets you interact with a given table (create it, query it, delete it, insert things into it, etc)

File: sdk/data_tables/src/clients/entity_client.rs [hunk 2]
```diff
@@ -9,11 +9,11 @@ use std::sync::Arc;
 use url::Url;
 
 pub trait AsEntityClient<RK: Into<String>> {
```
[the remark this answers is not in the document]
Comment (bmc-msft) [reply sdk/data_tables/src/clients/entity_client.rs:11]: Agreed.  These traits add an unnecessary layer of abstraction that reduces discovery of the functionality.

[the remark this answers is not in the document]
Comment (bmc-msft) [reply sdk/data_tables/src/clients/entity_client.rs:11]: I will investigate removing these in a follow-up PR.

**Pull request merged**